libs/utils/src/postgres_connection.rs

```diff
@@ -0,0 +1,199 @@
+use anyhow::Context;
+use std::fmt;
+use url::Host;
+
+pub fn parse_host_port<S: AsRef<str>>(host_port: S) -> Result<(Host, Option<u16>), anyhow::Error> {
+    let url = url::Url::parse(&format!("postgres://{}", host_port.as_ref()))
+        .context("Invalid host:port pair specified")?;
+    let host = url.host().context("Host is not specified")?.to_owned();
+    Ok((host, url.port()))
+}
+
+#[cfg(test)]
+mod tests_parse_host_port {
+    use crate::postgres_connection::parse_host_port;
+    use url::Host;
+
+    #[test]
+    fn test_normal() {
+        let (host, port) = parse_host_port("hello:123").unwrap();
+        assert_eq!(host, Host::Domain("hello".to_owned()));
+        assert_eq!(port, Some(123));
+    }
+
+    #[test]
+    fn test_no_port() {
+        let (host, port) = parse_host_port("hello").unwrap();
+        assert_eq!(host, Host::Domain("hello".to_owned()));
+        assert_eq!(port, None);
+    }
+
+    #[test]
+    fn test_ipv6() {
+        let (host, port) = parse_host_port("[::1]:123").unwrap();
+        assert_eq!(host, Host::<String>::Ipv6(std::net::Ipv6Addr::LOCALHOST));
+        assert_eq!(port, Some(123));
+    }
+
+    #[test]
+    fn test_invalid() {
+        assert!(parse_host_port("hello world").is_err());
+    }
+}
+
+#[derive(Clone)]
+pub struct PgConnectionConfig {
+    host: Host,
+    port: u16,
+    password: Option<String>,
+    options: Vec<String>,
+}
+
+impl PgConnectionConfig {
+    pub fn new_host_port(host: Host, port: u16) -> PgConnectionConfig {
+        PgConnectionConfig {
+            host,
+            port,
+            password: None,
+            options: vec![],
+        }
+    }
+
+    pub fn host(&self) -> &Host {
+        &self.host
+    }
+
+    pub fn port(&self) -> u16 {
+        self.port
+    }
+
+    pub fn set_password(mut self, s: Option<String>) -> Self {
+        self.password = s;
+        self
+    }
+
+    pub fn add_options<I: IntoIterator<Item = S>, S: Into<String>>(mut self, i: I) -> Self {
+        for s in i {
+            self.options.push(s.into());
+        }
+        self
+    }
+
+    /// Return a `<host>:<port>` string.
+    pub fn raw_address(&self) -> String {
+        format!("{}:{}", self.host(), self.port())
+    }
+
+    /// Build a client library-specific connection configuration.
+    /// Used for testing and when we need to add some obscure configuration
+    /// elements at the last moment.
+    pub fn build_tokio_postgres_config(&self) -> tokio_postgres::Config {
```

Comment:
Most of the time rust's naming conventions use `to_` (ref to owned), `into_` (owned to owned) and `as_` (ref to ref), so this should be `to_tokio_postgres_config`.